### PR TITLE
CA-201728: Add new interface `is_vlan_in_use`

### PR DESCRIPTION
**Pull request by sharady**
Signed-off-by: sharad yadav <sharad.yadav@citrix.com>

### CODE REVIEW

**Comment by robhoes** [comment]
This function definition suggests that the VLAN is going to be identified by its common "ethx.y" device name. We should not do that, because this is an implementation detail of the kernel. Instead, we should identify the VLAN by its parent device name plus the integer VLAN tag (as two separate arguments to the function).

Also, I would put the function inside the `Interface` module, and call it `has_vlan`.

File: network/network_interface.ml [hunk 1]
```diff
@@ -203,6 +204,7 @@ module Interface = struct
 	external get_capabilities : debug_info -> name:iface -> string list = ""
 	external is_connected : debug_info -> name:iface -> bool = ""
 	external is_physical : debug_info -> name:iface -> bool = ""
+	external has_vlan: debug_info -> device:string -> vlan:int64 -> bool = ""
```
Comment (robhoes) [inline network/network_interface.ml]: Please use `name:iface` rather than `device`, like the other functions have.

Comment (robhoes) [inline network/network_interface.ml]: `vlan` does not need to be `int64`. The one in `bridge_config_t` uses just `int`.

**Comment by sharady** [comment]
@robhoes thanks for the review, I have updated the same.

**Comment by lindig** [comment]
What are the `string * string` values carried by the exception?

**Comment by sharady** [comment]
For showing which device and vlan tag is in use by kernel while raising exception.

**Comment by robhoes** [comment]
> For showing which device and vlan tag is in use by kernel while raising exception.

You can make the second argument an `int` then, just like in the RPC.

**Comment by sharady** [comment]
It was list because the exception was raised as text `["device","eth0"; "vlan","tag"]`
Its fine to have a `string * int`, I have updated the PR.

**Pull request merged**